src/overlays/actors/ovl_Boss_02/z_boss_02.c

```diff
@@ -97,186 +151,2144 @@ const ActorInit Boss_02_InitVars = {
+        if (effect->unk_24 == 2) {
+            if (!flag) {
+                gSPDisplayList(POLY_XLU_DISP++, D_06000230);
+                gDPSetEnvColor(POLY_XLU_DISP++, 30, 30, 30, 128);
+                flag++;
+            }
+
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 30, 30, 30, (u8)effect->unk_2C);
+            gSPSegment(POLY_XLU_DISP++, 0x08,
+                       Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, effect->unk_26 + (i * 3),
+                                        (effect->unk_26 + (i * 3)) * 5, 32, 64, 1, 0, 0, 32, 32));
+
+            SysMatrix_InsertTranslation(effect->unk_00.x, effect->unk_00.y, effect->unk_00.z, 0);
+            SysMatrix_NormalizeXYZ(&globalCtx->mf_187FC);
+            Matrix_Scale(effect->unk_34 * D_809DF5B0, effect->unk_34 * D_809DF5B0, 1.0f, MTXMODE_APPLY);
+
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, D_060002E0);
+        }
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+#ifdef NON_MATCHING
+// Small ordering in the last loop
+void func_809DD934(Boss02* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 i;
+    Actor* temp_a0_5;
+    Vec3f sp58;
+    u8 sp57 = 0;
+    f32 phi_f0_2;
+    s16 phi_v1;
+
+    this->unk_1D14++;
+
+    switch (this->unk_1D18) {
+        case 0:
+            if (player->stateFlags1 & 0x100) {
+                func_800EA0D4(globalCtx, &globalCtx->csCtx);
+                this->unk_1D22 = func_801694DC(globalCtx);
+                func_80169590(globalCtx, 0, 1);
+                func_80169590(globalCtx, this->unk_1D22, 7);
+                func_8016566C(150);
+                this->unk_1D14 = 0;
+                this->unk_1D5C = 0.0f;
+                this->unk_1D58 = 0.0f;
+                if (D_809E0422 == 0) {
+                    this->unk_1D18 = 1;
+                    this->unk_1D68 = 10.0f;
+                    this->unk_1D64 = 60.0f;
+                    this->unk_1D6C = 23.0f;
+                    this->unk_1D70 = 0.01f;
+                    this->unk_1D74 = 0.0f;
+                    goto label1;
+                } else {
+                    this->unk_1D18 = 10;
+                    this->unk_1D68 = 10.0f;
+                    this->unk_1D64 = 200.0f;
+                    this->unk_1D6C = 273.0f;
+                    this->unk_1D70 = 0.1f;
+                    this->unk_1D74 = -100.0f;
+                    sp57 = 1;
+                    goto label2;
+                }
+            }
+            break;
+
+        case 1:
+            if ((this->unk_1D14 < 80U) && (D_809E0420 != 0) && (globalCtx->state.input[0].press.button & 0xC00F)) {
+                this->unk_1D18++;
+                this->unk_1D78 = 1;
+                this->unk_1D14 = 0;
+            } else {
+            label1:
+                if (this->unk_1D14 >= 50U) {
```

Comment:
```suggestion
            if ((this->unk_1D14 < 80) && (D_809E0420 != 0) && (globalCtx->state.input[0].press.button & 0xC00F)) {
                this->unk_1D18++;
                this->unk_1D78 = 1;
                this->unk_1D14 = 0;
            } else {
            label1:
                if (this->unk_1D14 >= 50) {
```
Are these `U`s required? There's a lot of them in this function, I won't comment on any more, but applies to those as well

Reply:
Yeah the Us are required.